### PR TITLE
polls/models: add validation before save

### PR DESCRIPTION
**Pull request by Rineee**
Tests for new serializers are still missing, but there is a task in taiga and I wanted to wait until we are sure that this is the way to do it.. 

The serialized data can be seen by adding mixinx.RetrieveModelMixin to PollViewSet in polls/api.py . Then /api/polls/pollId shows all questions of the poll (although the vote annotations arent there, as the questions are only annotated in the VoteViewSet and not in the PollViewSet). I tested with Opin, hope it also works in mB :nerd_face:

### CODE REVIEW

File: adhocracy4/polls/models.py [hunk 1]
```diff
@@ -148,6 +163,23 @@ class Choice(models.Model):
...
+    def clean(self, *args, **kwargs):
+        if self.question.is_open:
+            raise ValidationError({
+                'question': _('Open questions cannot have choices.')
+            })
+        if self.is_other_choice and self.question.choices.count() == 0:
+            raise ValidationError({
+                'is_other_choice': _('"Other" cannot be the only choice. Use '
+                                     'open question or add another choice '
+                                     'first.')
```
Comment (fuzzylogic2000) [inline adhocracy4/polls/models.py:175]: The text sounds a bit like you are only allowed to check the box if there are questions added first, so maybe without the first?
'"Other" cannot be the only choice. Use '
'open question or add more choices.'

**Review by fuzzylogic2000 — COMMENTED** [review]
Looks good! (But the tests are failing...) And there are some missing. :)

File: adhocracy4/polls/models.py [hunk 2]
```diff
@@ -71,15 +71,21 @@ def clean(self, *args, **kwargs):
             if self.multiple_choice:
                 raise ValidationError({
                     'is_open': _('Questions with open answers cannot '
-                                 'have multiple choices')
+                                 'have multiple choices.')
                 })
-            elif self.has_other_option:
```
Comment (Rineee) [inline adhocracy4/polls/models.py:76]: This is now checked when adding a choice

**Comment by Rineee** [comment]
Hm, tests are broken, because apparently I now broke the question api.. will look into this tomorrow!

**Comment by Rineee** [comment]
Ok, tests are fixed, but tests for new serializers are missing, and also tests for both question and vote api should be updated to also test the open stuff.. will add a task and do that once everything is settled.

For voting, the api now expects data of the form:
`    data = {
        'choices': [choice1.pk, choice2.pk],
        'other_choice_answer': 'myOtherChoiceAnswer',
        'open_answer': 'myOpenAnswer'
    }
`

File: adhocracy4/polls/static/polls/PollManagement.jsx [hunk 3]
```diff
@@ -43,6 +43,7 @@ class PollManagement extends React.Component {
   */
 
   getNewQuestion (label = '') {
+    console.log('I am so here')
```
Comment (fuzzylogic2000) [inline adhocracy4/polls/static/polls/PollManagement.jsx:46]: This should probably also go before merge?

**Review by fuzzylogic2000 — APPROVED** [review]
It looks good and we can merge to get it all together and change the two things and more afterwards!

File: adhocracy4/polls/static/polls/PollManagement.jsx (hunk 3 above)
Comment (Rineee) [reply adhocracy4/polls/static/polls/PollManagement.jsx:46]: haha, ups :see_no_evil:

**Pull request merged**